pkg/koordlet/metricsadvisor/collectors/coldmemoryresource/cold_page_kidled.go

```diff
@@ -0,0 +1,238 @@
+
+	nodeColdPageInfoMetric, err := k.collectNodeColdPageInfo()
+	if err != nil {
+		klog.Warningf("generate node cold page info metrics failed, err %v", err)
+	}
+	coldPageMetrics = append(coldPageMetrics, nodeColdPageInfoMetric...)
+
+	podsColdPageInfoMetric, err := k.collectPodsColdPageInfo()
+	if err != nil {
+		klog.Warningf("generate pods or container cold page info metrics failed, err %v", err)
+	}
+	coldPageMetrics = append(coldPageMetrics, podsColdPageInfoMetric...)
+
+	appender := k.appendableDB.Appender()
+	if err := appender.Append(coldPageMetrics); err != nil {
+		klog.ErrorS(err, "Append node metrics error")
+		return
+	}
+
+	if err := appender.Commit(); err != nil {
+		klog.Warningf("Commit node metrics failed, reason: %v", err)
+		return
+	}
+
+	k.started.Store(true)
+	klog.V(4).Info("collectColdPageInfo finished")
+}
+
+func (k *kidledcoldPageCollector) collectNodeColdPageInfo() ([]metriccache.MetricSample, error) {
+	coldPageMetrics := make([]metriccache.MetricSample, 0)
+	collectTime := time.Now()
+	// /sys/fs/cgroup/memory/memory.idle_page_stats
+	path := filepath.Join(system.Conf.CgroupRootDir, system.CgroupMemDir, system.MemoryIdlePageStatsName)
+	coldPageInfo, err := koordletutil.KidledColdPageInfo(path)
+	if err != nil {
+		return nil, err
+	}
+	memUsageWithHotPageBytes, err := coldPageInfo.NodeMemWithHotPageUsageBytes()
+	if err != nil {
+		return nil, err
+	}
+	memUsageWithHotPageValue := float64(memUsageWithHotPageBytes)
+	memUsageWithHotPageMetrics, err := metriccache.NodeMemoryWithHotPageUsageMetric.GenerateSample(nil, collectTime, memUsageWithHotPageValue)
+	if err != nil {
+		return nil, err
+	}
+	coldPageMetrics = append(coldPageMetrics, memUsageWithHotPageMetrics)
+
+	nodeColdPageBytes := coldPageInfo.GetColdPageTotalBytes()
+	nodeColdPageBytesValue := float64(nodeColdPageBytes)
+	nodeColdPageMetrics, err := metriccache.NodeMemoryColdPageSizeMetric.GenerateSample(nil, collectTime, nodeColdPageBytesValue)
+	if err != nil {
+		return nil, err
+	}
+	coldPageMetrics = append(coldPageMetrics, nodeColdPageMetrics)
+	klog.V(4).Infof("collectNodeResUsed finished, count %v, memUsageWithHotPage[%v], coldPageSize[%v]",
+		len(coldPageMetrics), memUsageWithHotPageValue, nodeColdPageBytes)
+	return coldPageMetrics, nil
+}
+
+func (k *kidledcoldPageCollector) collectPodsColdPageInfo() ([]metriccache.MetricSample, error) {
+	podMetas := k.statesInformer.GetAllPods()
+	count := 0
+	coldMetrics := make([]metriccache.MetricSample, 0)
+	for _, meta := range podMetas {
+		pod := meta.Pod
+		uid := string(pod.UID) // types.UID
+		podKey := util.GetPodKey(pod)
+		if filtered, msg := k.FilterPod(meta); filtered {
+			klog.V(5).Infof("skip collect pod %s, reason: %s", podKey, msg)
+			continue
+		}
+		collectTime := time.Now()
+		podCgroupDir := meta.CgroupDir
+		// /host-sys/fs/cgroup/memory/podDir/memory.idle_page_stats
+		path := filepath.Join(system.Conf.CgroupRootDir, system.CgroupMemDir, podCgroupDir, system.MemoryIdlePageStatsName)
```

Comment:
please use the CgroupResource to generate the filepath. you can follow the other pod collectors